lisp/ess-s-lang.el

```diff
@@ -694,52 +658,18 @@ In that case, it is removed and replaced by `ess-smart-S-assign-key'.
           (delete-horizontal-space))
       (insert ess-S-assign))))
 
-;;; Setting / Unsetting the smart S-assign-key behavior -----------------
-
-;; Two basic building blocks, used below:
-(defun ess--unset-smart-S-assign-key ()
-  (define-key ess-mode-map          "_" nil)
-  (define-key inferior-ess-mode-map "_" nil)
-  (define-key ess-mode-map          ess-smart-S-assign-key nil); 'self-insert-command
-  (define-key inferior-ess-mode-map ess-smart-S-assign-key nil))
-(defun ess--activate-smart-S-assign-key ()
-  (define-key ess-mode-map          ess-smart-S-assign-key 'ess-smart-S-assign)
-  (define-key inferior-ess-mode-map ess-smart-S-assign-key 'ess-smart-S-assign))
-
-
-;; Written such that whimps can have (ess-disable-smart-S-assign) in .emacs :
-(defun ess-disable-smart-S-assign (activate)
-  "Disable or activate (if prefix argument ACTIVATE is set) the smart assignment
-operator `ess-S-assign'.  That, typically \" <- \", can be customized."
-  (interactive "P")
-  (if activate
-      (ess--activate-smart-S-assign-key)
-    (ess--unset-smart-S-assign-key)))
-(defalias 'ess-disable-smart-underscore 'ess-disable-smart-S-assign)
-
-(defun ess-toggle-S-assign (force)
-  "Set the `ess-smart-S-assign-key' (by default \"_\"
- [underscore]) key to \\[ess-smart-S-assign] or back to
-`ess-smart-S-assign-key'.  Toggle the current definition, unless
-FORCE is non-nil, where \\[ess-smart-S-assign] is set
-unconditionally.
-
-If you as per default have `ess-smart-S-assign-key' set to
-underscore, note that using \"C-q _\" will always just insert the
-underscore character."
-  (interactive "P")
-  (let ((current-key (lookup-key ess-mode-map ess-smart-S-assign-key))
-        (default-key (lookup-key ess-mode-map "_")))
-    (if (and (or default-key current-key)
-             ;; (stringp current-key) (string= current-key ess-S-assign)
-             (not force))
-        (ess--unset-smart-S-assign-key)
-      ;; else : "force" or current-key is "nil", i.e. default
-      (ess--activate-smart-S-assign-key))))
-(defalias 'ess-toggle-underscore 'ess-toggle-S-assign)
-;; NOTA BENE: "_" is smart *by default* :
-;; -----  The user can always customize `ess-S-assign' ...
-(ess-toggle-S-assign 'force-to-S-assign)
+(defalias 'ess--unset-smart-S-assign-key 'ignore "")
+(make-obsolete 'ess--unset-smart-S-assign-key
+               "it does nothing. Set `ess-smart-S-assign-key' to nil." "2018-06-08")
+(defalias 'ess-disable-smart-S-assign 'ignore "")
+(make-obsolete 'ess-disable-smart-S-assign
+               "it does nothing. Set `ess-smart-S-assign-key' to nil." "2018-06-08")
+(defalias 'ess--activate-smart-S-assign-key 'ignore "")
+(make-obsolete 'ess--activate-smart-S-assign-key
+               "it does nothing. Set `ess-smart-S-assign-key' instead." "2018-06-08")
+(defalias 'ess-toggle-S-assign 'ignore "")
+(make-obsolete 'ess-toggle-S-assign
```

Comment:
I like the change but this one should be setting ess-smart-S-assign-key to nil. Otherwise this will lead to a lot of breakage. This  "feature" is already annoying to a lot of users, let's not irritate them yet again on this occasion.

Reply:
By "this one" you mean `ess-toggle-S-assign` should actually just do something like `(setq ess-smart-S-assign-key nil)`, right? That's an easy enough change to make.